### ISSUE TITLE
TrailingCommaOnDeclarationSite adds a comma before ; in enums

### ISSUE THREAD

**Issue by alvr** [pinterest/ktlint#1714]
After updating detekt to v1.22.0 and its rules, `TrailingCommaOnDeclarationSite` is adding a trailing comma before the semicolon for enums. For example:

```kotlin
enum class Example {
    A,
    B;

    fun something() = "blah"
}

enum class Example2 {
    A,
    B;

    companion object {
        val values = listOf(A, B)
    }
}
```

is formatted into

```kotlin
enum class Example {
    A,
    B,
    ;

    fun something() = "blah"
}

enum class Example2 {
    A,
    B,
    ;

    companion object {
        val values = listOf(A, B)
    }
}
```

When the last item doesn't have a semicolon the format is fine.

The rules used (updated as per the documentation)

```yaml
TrailingCommaOnCallSite:
  active: true
  autoCorrect: true
  useTrailingCommaOnCallSite: true
TrailingCommaOnDeclarationSite:
  active: true
  autoCorrect: true
  useTrailingCommaOnDeclarationSite: true
```

Is it possible to disable the rule for the last item of an enum if it has a semicolon?

**Comment by alvr** [comment]
Just saw #1679 but even suppressing the error with `@Suppress("TrailingCommaOnDeclarationSite", "ktlint:trailing-comma-on-declaration-site")` is still being reported

**Comment by paul-dingemans** [comment]
The suppression also works with your examples when using `ktlint` itself. Given that, I think that the problem should be analyzed and resolved in detekt.
